pkg/sip/service.go

```diff
@@ -388,3 +394,49 @@ func (s *Service) validateCallProvider(state *CallState) error {
 
 	return nil
 }
+
+// extractTransferErrorReason extracts a user-friendly reason string from an error
+func extractTransferErrorReason(err error) string {
+	if err == nil {
+		return "unknown"
+	}
+
+	// Check for psrpc errors
+	var psrpcErr psrpc.Error
+	if errors.As(err, &psrpcErr) {
```

Comment:
We should probably check for the `livekit.SIPStatus` first before falling back to PSRPC error codes.